### ISSUE TITLE
Using plugin in a "receive only" mode w/o connection string

### ISSUE THREAD

**Issue by bledbet** [SeanFeldman/ServiceBus.AttachmentPlugin#24]
Hi - great work.  Appologies if I am missing something obvious - I have a working sample of a sender and reciever, but it is not quite clear to me how you configure the receiver such that it does not need the storage account connection string - I think I see in the code of the plugin itself that it may not leverage the connection string in the case that the message has the SAS uri, but the connection string still seems to be required in order to register the plugin with the MessageReceiver (connection string is a required parameter to AzureStorageAttachmentConfiguration constructor, and it looks like the plugin's constructor would blow if it weren't available, as it attaches to the account at that time).  Thanks!

**Comment by SeanFeldman** [comment]
Thank you @bledbet 🙂

Sending a message without exposing the storage account to receivers feature was built with an design assumption that receivers of the ASB message are not part of your system and would receive a message the way they can and retrieve the body the way then need to as well.

Saying that, I see what you're asking - a case where 2 parties agree on this plugin, but the sending party has the storage account connection string, while receiving party doesn't. Is that correct? For this scenario to exist, the ASB connection string _still_ has to be shared to retrieve messages. Or at least a SAS token needs to be provided. In such case, does it make sense to share ASB credential, but not to share the Storage Account connection string? I'm opened to hear and learn about scenario you have. If this is a valid scenario in the real world, it would make sense to add support for the case.

**Comment by bledbet** [comment]
Hey Sean - yeah, I think you have correctly understood what I was thinking.  I am implementing pub/sub in an enterprise, trying to apply domain separation, and try to limit coordination of external resources (connection strings) across domains, so I like that the SAS token approach lets the consumer receive everything necessary to receive the payload through the brokered message, and yes, was thinking both ends would use the library.
That said, I actually can't use your library right now, as I am still on the older service bus client assembly, and was mostly asking the question to contribute, in case this was an oversight.
Thanks again, though, as this will still help me accomplish what I need to accomplish.

**Comment by SeanFeldman** [comment]
@bledbet thinking about it, it's a legitimate use case.
If you want to contribute, feel free to send a PR. If you want to collaborate, we can do that over a PR as well.

The old client (`WindowsAzure.ServiceBus`) doesn't support a plugin model and on its way "out". Hence why it's not something I'll focus on.

**Comment by SeanFeldman** [comment]
How about API like this:

```c#
receiver.RegisterAzureStorageAttachmentPluginForRecievingOnly();
```

- Will not requried `AzureStorageAttachmentConfiguration` argument
- Will only attempt to use SAS URI if one exists
- Will throw if is used for sending (sender)

**Comment by bledbet** [comment]
That could work.  Just food for thought: In my own framework (because I can't take the time to refactor around the newer service bus client assembly right now), my producer and consumer are separate - you could separate your plugin into something like an "AttachmentProducerPlugin" and a "AttachmentConsumerPlugin", they could both just accept the configuration - the producer plugin would pretty much always need the connection string, and would need to throw a sensible exception if it's not provided on the config.  The consumer might need the connection string and might not.  I'm not sure if you are wanting a single consumer plugin that can deal with some incoming messages needing the connection string and some don't - I would think that a given consumer instance would either always need the connection string or would always not need it.  So if you wanted deal with both scenarios in one consumer plugin, you setup the ecapsulated storeage client if the connection string is provided, and then upon message processing, if the url isn't SAS, and the storage client isn't constructed, then again you have to throw.  You might be able to deal with all of that in a single plugin class, but I'm inclined to separate them.

**Comment by SeanFeldman** [comment]
That's more or less the intent behind [my comment here](https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/issues/24#issuecomment-346550205). Extension method just hides away the details from the callers that don't really care if it's the same plugin or two different ones 🙂 

I'll probably add the feature as it seems very reasonable given the scenario you've provided. Thank you @bledbet.

**Comment by SeanFeldman** [comment]
@bledbet released as `2.1.0`.
Thanks for a good idea.
